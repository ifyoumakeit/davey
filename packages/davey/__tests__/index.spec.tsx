import { davey } from "../index";

describe("Basic single element", () => {
  const Component = <h1>hello</h1>;
  test("match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });
  test("returns tuple", () => {
    expect(Array.isArray(Component)).toBe(true);
  });
  test("first element of tuple is tag", () => {
    expect(Component[0]).toBe("h1");
  });
  test("second element of tuple is props object", () => {
    expect(typeof Component[1]).toBe("object");
  });
  test("second element of tuple has children", () => {
    expect(Component[1]).toHaveProperty("children");
  });
  test("children are an array", () => {
    expect(Array.isArray(Component[1].children)).toBe(true);
  });
  test("children contain correct text string", () => {
    expect(Component[1].children[0]).toBe("hello");
  });
});

describe("2-level deep nested element", () => {
  const Component = (
    <main>
      <header>
        <h1>hello</h1>
        <h2>world</h2>
      </header>
    </main>
  );
  test("match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });
  test("returns tuple", () => {
    expect(Array.isArray(Component)).toBe(true);
  });
  test("first element of tuple is tag", () => {
    expect(Component[0]).toBe("main");
  });
  test("second element of tuple is props object", () => {
    expect(typeof Component[1]).toBe("object");
  });
  test("second element of tuple has children", () => {
    expect(Component[1]).toHaveProperty("children");
  });
  test("children are an array", () => {
    expect(Array.isArray(Component[1].children)).toBe(true);
  });
  test("children contain next child node", () => {
    expect(Component[1].children[0][0]).toBe("header");
  });
  test("hello world be present", () => {
    expect(Component[1].children[0][1].children[0][1].children[0]).toBe("hello");
    expect(Component[1].children[0][1].children[1][1].children[0]).toBe("world");
  });
});
