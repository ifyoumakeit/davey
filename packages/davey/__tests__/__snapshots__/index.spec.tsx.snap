// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2-level deep nested element match snapshot 1`] = `
Array [
  "main",
  Object {
    "children": Array [
      Array [
        "header",
        Object {
          "children": Array [
            Array [
              "h1",
              Object {
                "children": Array [
                  "hello",
                ],
              },
            ],
            Array [
              "h2",
              Object {
                "children": Array [
                  "world",
                ],
              },
            ],
          ],
        },
      ],
    ],
  },
]
`;

exports[`Basic single element match snapshot 1`] = `
Array [
  "h1",
  Object {
    "children": Array [
      "hello",
    ],
  },
]
`;
